sys/kern/kern_malloc.c

```diff
@@ -389,6 +387,11 @@ malloc_type_zone_allocated(struct malloc_type *mtp, unsigned long size,
 	if (size > 0) {
 		mtsp->mts_memalloced += size;
 		mtsp->mts_numallocs++;
+#ifdef CHERI_PURECAP_KERNEL
```

Comment:
I would also prefer using `__CHERI_PURE_CAPABILITY__` for new code here and in other places.

Reply:
My plan is to keep them until the merge with dev and make sure that we correctly annotate the changes so that we keep track for analysis purposes.

Reply:
Any .c file in sys though is unambiguous when it comes to `__CHERI_PURE_CAPABILITY__`, it's only headers where you need to distinguish between kernel and userspace.